src/driver/argument_parser.hpp

```diff
@@ -590,6 +590,12 @@ struct argument_parser
                 v(*this, inputs);
             return arg.action(*this, inputs);
         }
+        catch(const std::runtime_error& e)
+        {
+            msg = e.what();
+            std::cerr << "error: " << msg << std::endl;
+            return false;
+        }
         catch(const std::exception& e)
         {
             msg = e.what();
```

Comment:
This clause is simply storing the exception type. Effectively ignoring it..

Reply:
It doesn't ignore. We print the message from the exception below:

```cpp
        try
        {
            ...
        }
        catch(const std::exception& e)
        {
            msg = e.what();
        }
        catch(...)
        {
            msg = "unknown exception";
        }
        std::cout << color::fg_red << color::bold << "error: " << color::reset;
        auto sc = spellcheck(inputs);
        if(sc.distance < 5)
        {
            ...
        }
        else
        {
            const auto& flag_name = flag.empty() ? arg.metavar : flag;
            std::cout << "Invalid input to '" << color::fg_yellow;
            std::cout << arg.usage(flag_name);
            std::cout << color::reset << "'" << std::endl;
            std::cout << "       " << msg << std::endl; // <- Print the error message from the exception
            std::cout << std::endl;
            print_usage({arg.usage()});
        }
```

We then return `true` which means to not continue executing the command.